src/main/java/fruitshop/service/writer/FileWriterImpl.java

```
@@ -0,0 +1,13 @@
+package fruitshop.service.writer;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+public class FileWriterImpl implements FileWriter {
+    @Override
+    public void write(String data, String fileName) throws IOException {
```

This violates checklist item: 'Don't ignore exceptions in try-catch blocks'. The method throws IOException directly, but the requirements specify that IOExceptions should be caught and rethrown as RuntimeExceptions with informative messages and the original cause.